api/v1alpha1/service_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2025 The ORC Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// ServiceResourceSpec contains the desired state of the resource.
+type ServiceResourceSpec struct {
+	// type indicates which resource the service is responsible for.
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=255
+	// +optional
+	Type *string `json:"type,omitempty"`
+
+	// enabled indicates whether the service is enabled or not.
+	// +kubebuilder:default=true
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// extra indicates key-value information about the service.
+	Extra map[string]any `json:"extra,omitempty"`
```

Comment:
That appears to be legacy, undocumented API. I'd recommend we stick with what's documented at https://docs.openstack.org/api-ref/identity/v3/#create-service.

@dlaw4608 had the same problem with the Role controller, and he ended up updating gophercloud to properly support the missing fields (so that we can eventually deprecate Extra in gophercloud). Have a look at https://github.com/gophercloud/gophercloud/pull/3532.